### DISCUSSION TITLE
Mutually exclusive component options

### DISCUSSION THREAD

**Discussion by RothAndrew** [zarf-dev/zarf#70] [Ideas]
Right now the only options for a component are `yes` or `no`. What do you think about having the ability to do a mutually exclusive choice? The use case I'm thinking about is being able to use the exact same zarf package for a local deployment as you would for production, but there are a couple of tweaks that have to be made.

Example:

```
Do you want a local or production deployment? [local|prod]
local
Okay! Setting `domain` to `bigbang.dev`
```

```
Do you want a local or production deployment? [local|prod]
prod
Okay! Setting `domain` to `foo.com`
```

**Comment by RothAndrew** [comment]
Alternatives:

- Modifying my hosts file
- ???

**Comment by jeff-mccoy** [comment]
hmmm I'm confused a little on this one.  What data exactly would be changing within zarf?  There is the hostname prompt on `zarf init`, but that's just for the traefik TLS secret and can easily be changed.

**Reply by RothAndrew** [reply]
I'm thinking of packages to deploy, like a big bang deployment. If I'm local I'm going to want `bigbang.dev` and the well known TLS cert. If I'm using the same package to prod the domain and cert has to change. Just trying to figure out whether there is any way to have a dev use the same package for local dev/int as is used in prod

**Comment by RothAndrew** [comment]
Having Zarf set values for helm charts would probably not be the best course of action, but maybe having it choose between two files to package based on the choice?

**Comment by RothAndrew** [comment]
Right now I'm just spitballing after a cocktail or two, just wanted to get a conversation going. I'm totally willing to concede that it is a stupid idea :)

**Comment by jeff-mccoy** [comment]
Lol.  Oh so what you're generally saying is.... right now component options are additive by nature (AND):

Deploy A?
Deploy B?
Deploy C?

you're suggesting something like (OR):

Pick one:
A, B, C

?

**Reply by RothAndrew** [reply]
Yes, but I'd want the ability to pick between the two modes per component.

```
Deploy A? [Y/n]
Deploy B or C? [B/c]
Deploy D? [Y/n]
```

**Reply by jeff-mccoy** [reply]
Okay that's interesting, will need to think on how to do that one
